proof/archive.go

```diff
@@ -302,6 +298,12 @@ func (m *MultiArchiver) ImportProofs(ctx context.Context,
 
 			proof.ScriptKey = *finalAsset.ScriptKey.PubKey
 		}
+
+		return nil
+	}
+
+	if err := chanutils.ErrGroup(f, proofs); err != nil {
```

Comment:
We should thread through the context here so cancelling the context will cause all the entire thing to bail our early. 

We can use `errgroup.WithContext` above.